inc/ctest_2_cppunittest.h

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef CTEST_2_CPPUNITTEST_H
+#define CTEST_2_CPPUNITTEST_H
+
+#include "macro_utils/macro_utils.h"
+
+// This structure matches 100% the structure used by CppUnitTest to store its metadata in the shared section where the metadata is stored
+typedef struct CTEST_2_CPPUNITTEST_MethodMetadata_TAG
+{
+    const wchar_t* tag;
+    const wchar_t* methodName;
+    const unsigned char* helpMethodName;
+    const unsigned char* helpMethodDecoratedName;
+    const wchar_t* sourceFile;
+    int lineNo;
+} CTEST_2_CPPUNITTEST_MethodMetadata;
+
+typedef struct CTEST_2_CPPUNITTEST_MemberMethodInfo_TAG
+{
+    enum CTEST_2_CPPUNITTEST_MemberMethodTypes
+    {
+        TestMethod,
+        TestMethodSetup,
+        TestMethodCleanup,
+        TestObjectSetup,
+        TestObjectCleanup,
+        TestClassSetup,
+        TestClassCleanup,
+    } methodType;
+    union
+    {
+        void* pVoidMethod;
+        void* pVoidStaticMethod;
+    } method;
+
+    const CTEST_2_CPPUNITTEST_MethodMetadata* metadata;
+} CTEST_2_CPPUNITTEST_MemberMethodInfo;
+
+// copied from CppUnitTest, do not have a better solution at the moment
+#define ALLOCATE_TESTDATA_SECTION_VERSION __declspec(allocate("testvers$"))
+#define ALLOCATE_TESTDATA_SECTION_CLASS __declspec(allocate("testdata$_A_class"))
+#define ALLOCATE_TESTDATA_SECTION_METHOD __declspec(allocate("testdata$_B_method"))
+#define ALLOCATE_TESTDATA_SECTION_ATTRIBUTE __declspec(allocate("testdata$_C_attribute"))
+#define ALLOCATE_TESTDATA_SECTION_DAN __declspec(allocate("testdata$_dan"))
```

Comment:
>ALLOCATE_TESTDATA_SECTION_DAN [](start = 8, length = 29)

lol

Reply:
Haha, we will remove the dan

Reply:
Removed

---
In reply to: [608144209](https://github.com/Azure/c-testrunnerswitcher/pull/65#discussion_r608144209) [](ancestors = 608144209)